rust/private/rustc.bzl

```diff
@@ -1095,6 +1095,9 @@ def construct_arguments(
     if _is_no_std(ctx, toolchain, crate_info):
         rustc_flags.add('--cfg=feature="no_std"')
 
+    # Needed for bzlmod-aware runfiles resolution.
+    env["REPOSITORY_NAME"] = ctx.label.workspace_name
```

Comment:
Why wasn't the `workspace_name.env` file sufficient? Have other projects been adding `REPOSITORY_NAME` for runfiles support? I'd like to avoid adding things to the `Rustc` action if possible.

Reply:
If my understanding is correct, that file always encodes the rules_rust repo name. What we want is the repo_name of the calling code. So if the `_main` repo depends on `foo` which uses runfiles, it should be the mapped name for `foo` when code in `foo` looks up runfiles.

I checked the python and Go implementations and they use reflection/stack-walking to figure out the caller. Java [uses an annotation](https://github.com/bazelbuild/bazel/blob/master/tools/java/runfiles/Runfiles.java#L61-L74), C++ allows you to pass it into the constructor at creation time https://github.com/bazelbuild/bazel/blob/5594166dcbcdf746117ee4106ff5b6679434016c/tools/cpp/runfiles/runfiles_src.h#L129-L133

Reply:
@dzbarsky is right here, this kind of magic is indeed required to get the correct lookup behavior. I looked into this a bit more a while back and couldn't find any way to do this without `Rustc` changes.

Reply:
This looks good to me. I spent multiple hours trying to understand how the old one worked about a year ago, and never did work out how - turns out my understanding was correct and it was never working.